src/docfx/template/TemplateEngine.cs

```diff
@@ -5,34 +5,37 @@
 using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.IO;
-using System.Threading;
 using System.Threading.Tasks;
 using Newtonsoft.Json.Linq;
 
 namespace Microsoft.Docs.Build
 {
     internal class TemplateEngine
     {
+        public TemplateEngineRunner TemplateEngineRunner { get; init; }
```

Comment:
Consider to make this private and keep the methods for normal build usage to keep the existing build pipeline.
Because it might be refactored this further later within DocFX.

Reply:
For `TemplateModelBuilder` depends on `TemplateEngineRunner`, the current is better.